libs/common/src/vault/services/sync/sync.service.ts

```diff
@@ -145,6 +145,13 @@ export class SyncService extends CoreSyncService {
     }
 
     const response = await this.apiService.getAccountRevisionDate();
+    if (response < 0) {
```

Comment:
suggestion: I think we can avoid the nested ifs 

```suggestion
    if (response < 0 && this.logoutCallback) {
          // Account was deleted, log out now
          await this.logoutCallback(false);
    }
```